include/AutomatableModelView.h

```diff
@@ -97,10 +104,37 @@ class AutomatableModelViewSlots : public QObject
 public slots:
 	void execConnectionDialog();
 	void removeConnection();
+	void addSongAutomationNode();
+	void addSongAutomationNodeAndClip();
+	void updateSongNearestAutomationNode();
+	void removeSongNearestAutomationNode();
 	void editSongGlobalAutomation();
 	void unlinkAllModels();
 	void removeSongGlobalAutomation();
 
+private:
+	struct AutomationNodeAtTimePos
+	{
+		TimePos m_position;
+		AutomationClip* m_clip;
```

Comment:
```suggestion
		AutomationClip* clip = nullptr;
		TimePos position;
```
For simple structs, it's best to omit the "m_" prefix on data members.